cdap-proto/src/main/java/co/cask/cdap/proto/BusinessMetadataRecord.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.proto;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represent Busineess Metadata for CDAP.
+ */
+public class BusinessMetadataRecord {
+  private final String targetType;
+  private final Id.NamespacedId  targetId;
+  private final String key;
+  private final String value;
+  private final String schema;
+
+  public BusinessMetadataRecord(String targetType, Id.NamespacedId targetId, String key, String value) {
+    this(targetType, targetId, key, value, null);
+  }
+
+  public BusinessMetadataRecord(String targetType, Id.NamespacedId  targetId, String key, String value,
```

Comment:
It might be better to have `targetType` determined in this class based on `targetId`, rather than getting it as a parameter so that there is only one place to determine `targetType`.